src/extension/executors/runner.ts

```diff
@@ -10,22 +10,25 @@ import {
   TaskPanelKind,
   tasks,
   NotebookCellExecution,
-  TextDocument
+  TextDocument,
+  ExtensionContext
 } from 'vscode'
 
 import { OutputType } from '../../constants'
 import { CellOutputPayload } from '../../types'
 import { PLATFORM_OS } from '../constants'
 import { IRunner, IRunnerEnvironment } from '../runner'
-import { getAnnotations, getCmdShellSeq } from '../utils'
+import { getAnnotations, getCmdSeq, getCmdShellSeq } from '../utils'
 
 import { closeTerminalByEnvID } from './task'
-import { getShellPath } from './utils'
+import { getCommandExportExtractMatches, getShellPath, promptUserForVariable } from './utils'
 
 const LABEL_LIMIT = 15
+const BACKGROUND_TASK_HIDE_TIMEOUT = 2000
 const MIME_TYPES_WITH_CUSTOM_RENDERERS = ['text/plain']
 
 export async function executeRunner(
+  context: ExtensionContext,
```

Comment:
Side note: one of the reasons why I think this the function-abstraction of Executors won't suffice long-term. Not part of this PR, though.